### PR TITLE
intravisit: Fold functionality of IdVisitor into the regular Visitor.

### PR DESCRIPTION
**Pull request by michaelwoerister**

### CODE REVIEW

File: src/librustc/hir/intravisit.rs [hunk 1]
```diff
@@ -814,6 +861,7 @@ pub fn walk_arm<'v, V: Visitor<'v>>(visitor: &mut V, arm: &'v Arm) {
 }
 
 pub fn walk_vis<'v, V: Visitor<'v>>(visitor: &mut V, vis: &'v Visibility) {
+    // FIXME(mw): What is this NodeId in there? And should it be visited?
```
Comment (petrochenkov) [inline src/librustc/hir/intravisit.rs]: The path in visibility `pub(path)` has a resolution in `def_map` keyed by this id.

Comment (michaelwoerister) [inline src/librustc/hir/intravisit.rs]: OK, then we should visit it.

**Comment by alexcrichton** [comment]
@bors: rollup

assuming this isn't very platform-specific and green travis means it's good to go whenever

**Pull request merged**